mempool/src/core_mempool/transaction_store.rs

```diff
@@ -140,9 +140,8 @@ impl TransactionStore {
                         self.index_remove(&txn);
                     }
                 } else {
-                    return MempoolStatus::new(MempoolStatusCode::InvalidUpdate).with_message(
-                        format!("Failed to update gas price to {}", txn.get_gas_price()),
-                    );
+                    return MempoolStatus::new(MempoolStatusCode::InvalidUpdate)
+                        .with_message("Transaction already in mempool".to_string());
```

Comment:
From what I understand, mempool allows transactions to be resubmitted as long as the gas price of the transaction has increased. If not, we get here. Perhaps a better error message might be something like, `The transaction is already in mempool. Transactions can only be resubmitted if they increase the gas price.` (or something shorter, but to the same effect 😆 )

Reply:
Yeah, the only thing that can be resubmitted is a higher gas price.  I think `Transaction already in mempool` is sufficient, because if you resubmitted it with a lower gas price... It's kinda your fault haha.  If you're submitting it with the same gas price, then that's the appropriate message.

Reply:
Personally, I don't think that's clear enough because it assumes the user knows this behaviour (which may not be the case) but, either way, this is an improvement 😄

Reply:
@JoshLind thanks for catching the unit test failing. I fixed the `test_post_transaction_rejected_by_mempool` but there seems to be another test `test_bootstrap_jsonprc_and_api_configured_at_different_port` that is failing that seems to have nothing to do with my changes? Any chance you encountered this before?

Reply:
@hariria, it looks good now, right? 😄 Weird..

Reply:
If its failing randomly make sure you're fully rebased

Reply:
hmmm weird, i am fully rebased, ran unit tests again and everything passed.